core/dbt/contracts/graph/semantic_manifest.py

```diff
@@ -58,6 +60,19 @@ def validate(self) -> bool:
         semantic_manifest = self._get_pydantic_semantic_manifest()
         validator = SemanticManifestValidator[PydanticSemanticManifest]()
         validation_results = validator.validate_semantic_manifest(semantic_manifest)
+        new_time_spines = semantic_manifest.project_configuration.time_spines
+        old_time_spines = semantic_manifest.project_configuration.time_spine_table_configurations
+        new_time_spines_contain_day = any(
+            c for c in new_time_spines if c.primary_column.time_granularity == TimeGranularity.DAY
+        )
+        if (
```

Comment:
Not related to this PR, but seeing this makes me realized that some of our older parsing errors are likely in the wrong place and should be moved here! i.e., all the parsing errors raised in `_get_pydantic_semantic_manifest()`, which disrupt the parsing flow.  Maybe we can have a task to fix that after this one is finished.

Reply:
Actually, ignore this. Just looked into it more and I think we are actually following the same logic for how most core objects are validated